amundsen-kube-helm/templates/helm/amundsen/templates/deployment.yaml

```diff
@@ -103,16 +129,40 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
       {{- end }}
+      volumes:
+      {{- if .Values.oidcEnabled }}
+        - name: oidc-config
+          secret:
+            secretName: oidc-config
+      {{- end }}
       containers:
       - name: {{ .Chart.Name }}-{{ .Values.frontEndServiceName }}
-        image: {{- if .Values.frontEndServiceImage }} {{.Values.frontEndServiceImage}}{{- else }} {{ .Values.dockerhubImagePath }}/{{ .Chart.Name }}-{{ .Values.frontEndServiceName }}:{{ .Values.frontEndImageVersion }}{{- end }}
+        {{- with .Values.frontEndServiceImage }}
+        image: {{ . }}
+        {{- else }}
+        image: {{ .Values.dockerhubImagePath }}/{{ .Chart.Name }}-{{ .Values.frontEndServiceName }}{{ if .Values.oidcEnabled }}-oidc{{ end }}:{{ .Values.frontEndImageVersion }}
+        {{- end }}
         ports:
-        - containerPort: 5000  
+          - containerPort: 5000
         env:
-        - name: SEARCHSERVICE_BASE
-          value: http://{{ .Chart.Name }}-{{ .Values.searchServiceName }}:5001
-        - name: METADATASERVICE_BASE
-          value: http://{{ .Chart.Name }}-{{ .Values.metadataServiceName }}:5002
-        - name: FRONTEND_SVC_CONFIG_MODULE_CLASS
-          value: amundsen_application.config.TestConfig                     
----                          
+          - name: SEARCHSERVICE_BASE
+            value: http://{{ .Chart.Name }}-{{ .Values.searchServiceName }}:5001
+          - name: METADATASERVICE_BASE
+            value: http://{{ .Chart.Name }}-{{ .Values.metadataServiceName }}:5002
+          - name: LONG_RANDOM_STRING
+            value: {{ quote .Values.LONG_RANDOM_STRING }}
+        {{- if .Values.oidcEnabled }}
+          - name: FLASK_OIDC_CLIENT_SECRETS
+            value: /etc/client_secrets.json
+          - name: FLASK_OIDC_SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: oidc-config
+                key: OIDC_CLIENT_SECRET
+        {{- end }}
+        volumeMounts:
```

Comment:
If oidcEnabled=false, won't that leave the volumeMounts section in place and empty? I don't know if that is valid or not. Did you try running the helm template command to test out your changes (I don't know if there is another way to test it out)?

Reply:
Sorry for the delay. It is valid. 

It is a pretty common pattern. I could put the if statement around the entire block, but then if someone else was adding a second volume, they would have to have multiple if statements and it gets messy really quick. (example: https://github.com/helm/charts/blob/master/stable/artifactory/templates/nginx-deployment.yaml#L148-L170 If they put a if statement around the whole thing, it would be a pretty big statement. )

I ran this: `helm install amundsen -n amundsen --set oidcEnabled=false`
then kubectl describe deployment amundsen-frontend 
Everything works and in that output I see: 
`  Volumes:                               <none>`

Reply:
Works for me. Thanks.